### PR TITLE
feat(middleware): aws lambda middleware

### PR DESCRIPTION
**Pull request by leifermendez**
# Overview

# What is it?

- [ ] Feature / enhancement
- [ ] Bug
- [ ] Docs / tests / types / typos

# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

# Use cases and why

- 1. One use case
- 2. Another use case

# Checklist:

- [ ] My code follows the [developer guidelines of this project](https://github.com/BuilderIO/qwik/blob/main/CONTRIBUTING.md)
- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] Added new tests to cover the fix / functionality

### CODE REVIEW

File: starters/adapters/aws-lambda/src/entry_aws-lambda.tsx [hunk 1]
```diff
@@ -1,68 +1,22 @@
 /*
  * WHAT IS THIS FILE?
  *
- * It's the entry point for the Express HTTP server when building for production.
+ * It's the entry point for Vercel Edge when building for production.
  *
- * Learn more about Node.js server integrations here:
- * - https://qwik.builder.io/docs/deployments/node/
+ * Learn more about the Vercel Edge integration here:
```
Comment (zanettin) [inline starters/adapters/aws-lambda/src/entry_aws-lambda.tsx:6]: shouldn't it be aws? 🤔 😃

File: packages/qwik-city/middleware/aws-lambda/index.ts [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+      },
+      getOrigin(req) {
+        if (process.env.IS_OFFLINE) {
+          return `http://${req.headers.host}`;
+        }
+        return null;
+      },
+    });
+
+    const qwikApp = serverless(
```
Comment (albertshala1) [inline packages/qwik-city/middleware/aws-lambda/index.ts:26]: Hi @leifermendez while testing out this change, i'm seeing an issue with intercepting query parameters on the main route '/', where I can't query is always null, but the pathname is interceptable, wondering if there's something else that's needed here to address this.

**Comment by zanettin** [comment]
Hi @leifermendez 👋 
just wanted to quickly check, what the state of this PR is 👼 thx again for adding this integration 🙏

**Comment by leifermendez** [comment]
Hi! I will fix the conflicts and send again  ;) thanks 🚀

**Comment by gioboa** [comment]
I tried the integration but it's not working.
Screenshot 2023-11-01 at 16 32 39

**Comment by mhevery** [comment]
I would love to land this. What is still missing?

**Comment by PatrickJS** [comment]
@leifermendez any updates we want to merge this asap 🙏

**Comment by wmertens** [comment]
@leifermendez anything we can do to help this PR?

**Comment by leifermendez** [comment]
@wmertens I'm going to send a new PR because I don't know what went through my fork when sync went haywire.

**Pull request closed without merging**